### PR TITLE
Media Links Initial Commit

### PR DESCRIPTION
**Pull request by kathenry**
Co-Authored-By: Oliver Hamuy <o.hamuy@wustl.edu>
Co-Authored-By: Lucas Alcantara <lukmin1999@users.noreply.github.com>

### CODE REVIEW

File: rails/app/dashboards/place_dashboard.rb [hunk 1]
```diff
@@ -36,6 +37,7 @@ class PlaceDashboard < Administrate::BaseDashboard
     :lat,
     :stories,
     :photo,
+    # :media_links
```
Comment (FeminismIsAwesome) [inline rails/app/dashboards/place_dashboard.rb:40]: remove comment plz

File: rails/app/dashboards/place_dashboard.rb [hunk 2]
```diff
@@ -66,6 +68,7 @@ class PlaceDashboard < Administrate::BaseDashboard
     :lat,
     :stories,
     :photo,
+    # :media_links,
```
Comment (FeminismIsAwesome) [inline rails/app/dashboards/place_dashboard.rb:71]: remove comment plz

File: rails/app/dashboards/speaker_dashboard.rb [hunk 3]
```diff
@@ -30,7 +31,8 @@ class SpeakerDashboard < Administrate::BaseDashboard
     :name,
     :birthdate,
     :birthplace,
-    :community
+    :community,
+    # :media_links,
```
Comment (FeminismIsAwesome) [inline rails/app/dashboards/speaker_dashboard.rb:35]: remove comment plz

File: rails/app/javascript/components/StoryList.jsx [hunk 4]
```diff
@@ -80,6 +82,25 @@ class StoryList extends Component {
...
+        return(
+          <li>
+            <h6>
+              video {linkNum + 1}
+            </h6>
+            <iframe width="100%" height="auto" src={media_link.url} frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen>
+
+            </iframe>
+          </li>
+          // <li>
```
Comment (FeminismIsAwesome) [inline rails/app/javascript/components/StoryList.jsx:96]: remove comment plz

File: rails/app/javascript/components/StoryList.jsx [hunk 5]
```diff
@@ -113,6 +134,22 @@ class StoryList extends Component {
             {story.media &&
               story.media.map(file => <StoryMedia file={file} doBustCache={bustCache} key={story.media.id} />)}
           </div>
+          <div className="container">
+            <h6>
+              Media Links
+            </h6>
+            <ol>
+              {/* {story.media_links.map(media_link => 
+                <li>
```
Comment (FeminismIsAwesome) [inline rails/app/javascript/components/StoryList.jsx:143]: remove comment plz

File: rails/app/javascript/components/StoryList.jsx [hunk 6]
```diff
@@ -80,6 +82,25 @@ class StoryList extends Component {
     const bustCache = () => {
       this.cache.clear(index, 0);
     }
+    const mediaLinkList = 
+      story.media_links.map( function (media_link, linkNum) {
```
Comment (FeminismIsAwesome) [inline rails/app/javascript/components/StoryList.jsx:86]: can you make this a function instead? React paradigm wise, this looks more like a render function.

File: rails/app/models/media_link.rb [hunk 7]
```diff
@@ -0,0 +1,8 @@
+class MediaLink < ApplicationRecord
+    belongs_to :story
+
+    def self.update_url(url)
+        url["watch?v="]="embed/"
+        # "watch?v=" ==> "embed/"
```
Comment (FeminismIsAwesome) [inline rails/app/models/media_link.rb:6]: woah. I think i'd want a test for this, as i'm not quite sure that behavior is what i expect

File: rails/db/migrate/20200712200141_create_media_links.rb [hunk 8]
```diff
@@ -0,0 +1,13 @@
+class CreateMediaLinks < ActiveRecord::Migration[5.2]
+  def change
+    create_table :media_links do |t|
+      t.string :url 
+      t.references :story
+      t.timestamps
+    end
+  end
+  # def update_url(url)
+  #   url["watch?v="]="embed/"
```
Comment (FeminismIsAwesome) [inline rails/db/migrate/20200712200141_create_media_links.rb:10]: remove commentz

File: rails/app/models/media_link.rb (hunk 7 above)
Comment (kathenry) [reply rails/app/models/media_link.rb:6]: @FeminismIsAwesome I don't think this was working anyways, I'll just remove it for now and we'll continue to work on it.

File: rails/app/javascript/components/StoryList.jsx [hunk 9]
```diff
@@ -5,6 +5,8 @@ import Filter from "./Filter";
 import Sort from "./Sort";
 import { List, AutoSizer, CellMeasurer, CellMeasurerCache } from "react-virtualized";
 
+// import ReactPlayer from "react-player"
```
Comment (FeminismIsAwesome) [inline rails/app/javascript/components/StoryList.jsx:8]: more comments

File: rails/app/dashboards/place_dashboard.rb [hunk 10]
```diff
@@ -19,6 +19,7 @@ class PlaceDashboard < Administrate::BaseDashboard
     type_of_place: Field::String,
     created_at: Field::DateTime,
     updated_at: Field::DateTime,
+    # media_links: Field::HasMany,
```
Comment (FeminismIsAwesome) [inline rails/app/dashboards/place_dashboard.rb:22]: comments

File: rails/app/dashboards/speaker_dashboard.rb [hunk 11]
```diff
@@ -17,6 +17,7 @@ class SpeakerDashboard < Administrate::BaseDashboard
     community: Field::String,
     created_at: Field::DateTime,
     updated_at: Field::DateTime,
+    # media_links: Field::HasMany,
```
Comment (FeminismIsAwesome) [inline rails/app/dashboards/speaker_dashboard.rb:20]: comments

**Pull request merged**